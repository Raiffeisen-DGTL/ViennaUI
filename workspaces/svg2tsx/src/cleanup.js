const SVGO = require('svgo');

/**
 * Create instance SVGO
 */
const svgo = new SVGO({
    full: true,
    plugins: [
        { cleanupAttrs: true },
        { removeDoctype: true },
        { removeXMLProcInst: true },
        { removeComments: true },
        { removeMetadata: true },
        { removeTitle: true },
        { removeDesc: true },
        { removeUselessDefs: true },
        { removeEditorsNSData: true },
        { removeEmptyAttrs: true },
        { removeHiddenElems: true },
        { removeEmptyText: true },
        { removeEmptyContainers: true },
        { removeViewBox: false },
        { cleanupEnableBackground: true },
        { convertStyleToAttrs: true },
        { convertColors: true },
        { convertPathData: true },
        { convertTransform: true },
        { removeUnknownsAndDefaults: true },
        { removeNonInheritableGroupAttrs: true },
        { removeUselessStrokeAndFill: true },
        { removeUnusedNS: true },
        { cleanupIDs: true },
        { cleanupNumericValues: true },
        { cleanupListOfValues: true },
        { moveElemsAttrsToGroup: true },
        { moveGroupAttrsToElems: true },
        { collapseGroups: true },
        { removeRasterImages: true },
        { mergePaths: true },
        { convertShapeToPath: true },
        { convertEllipseToCircle: true },
        { sortAttrs: true },
        { removeDimensions: false },
        { sortAttrs: true },
        { sortDefsChildren: true },
        { reusePaths: true },
        { minifyStyles: true },
        { inlineStyles: true },
        { removeXMLNS: false },
    ],
});

/**
 * Clean up SVG by content
 *
 * @param {string} svg SVG string
 * @returns {string} Returns
 */
module.exports = async function cleaner(svg) {
    try {
        return await svgo.optimize(svg);
    } catch (error) {
        return error;
    }
};
