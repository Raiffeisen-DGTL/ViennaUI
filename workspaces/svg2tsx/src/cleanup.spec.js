import cleanup from './cleanup';

test('Cleanup', async () => {
    const content = `<svg width="17" height="17" viewBox="0 0 17 17" fill="none" xmlns="http://www.w3.org/2000/svg">
        <path d="M7.33541 5.22673L11.3365 5.22673L11.3365 6.56044H7.33541V5.22673Z" fill="#2B2D33"/>
        <path d="M4.66799 9.22787L8.66912 9.22787V7.89416H4.66799L4.66799 9.22787Z" fill="#2B2D33"/>
        <path d="M8.66912 11.8953H7.33541V10.5616H8.66912V11.8953Z" fill="#2B2D33"/>
        <path d="M4.66799 11.8953H6.0017V10.5616H4.66799V11.8953Z" fill="#2B2D33"/>
        <path d="M4.66799 5.22673H6.0017V6.56044H4.66799V5.22673Z" fill="#2B2D33"/>
        <path d="M10.0028 9.22787H11.3365V7.89416H10.0028V9.22787Z" fill="#2B2D33"/>
        <path fill-rule="evenodd" clip-rule="evenodd" d="M2.66742 1.89246C1.93083 1.89246 1.33371 2.48958 1.33371 3.22617V13.8959C1.33371 14.6324 1.93083 15.2296 2.66742 15.2296H13.3371C14.0737 15.2296 14.6708 14.6324 14.6708 13.8959V3.22617C14.6708 2.48958 14.0737 1.89246 13.3371 1.89246H2.66742ZM13.3371 3.22617H2.66742V13.8959H13.3371V3.22617Z" fill="#2B2D33"/>
        </svg>
    `;

    const svgo = await cleanup(content);

    expect(svgo.data).toMatchSnapshot();
    expect(svgo.info).toEqual({ width: '17', height: '17' });
});
