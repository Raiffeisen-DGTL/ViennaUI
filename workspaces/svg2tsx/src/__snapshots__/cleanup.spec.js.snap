// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cleanup 1`] = `"<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"17\\" height=\\"17\\" fill=\\"none\\" viewBox=\\"0 0 17 17\\"><defs/><path fill=\\"#2B2D33\\" d=\\"M7.335 5.227h4.001V6.56h-4V5.227zM4.668 9.228h4.001V7.894H4.668v1.334zM8.67 11.895H7.334v-1.333H8.67v1.333zM4.668 11.895h1.334v-1.333H4.668v1.333zM4.668 5.227h1.334V6.56H4.668V5.227zM10.003 9.228h1.333V7.894h-1.333v1.334z\\"/><path fill=\\"#2B2D33\\" fill-rule=\\"evenodd\\" d=\\"M2.667 1.892c-.736 0-1.333.598-1.333 1.334v10.67c0 .736.597 1.334 1.333 1.334h10.67c.737 0 1.334-.598 1.334-1.334V3.226c0-.736-.597-1.334-1.334-1.334H2.667zm10.67 1.334H2.667v10.67h10.67V3.226z\\" clip-rule=\\"evenodd\\"/></svg>"`;
